internal/user_validator.go

```diff
@@ -88,26 +92,99 @@ func (i *ValidateUser) Setup(ctx context.Context) error {
 	return nil
 }
 
-func decodePgpKey(pgpKey string) (*openpgp.Entity, error) {
+// crc24 calculates the CRC24 checksum for a given byte array.
+//
+// See the RFC 4880, "OpenPGP Message Format", Section 6.1, for source of this implementation.
+func crc24(bytes []byte) uint32 {
```

Comment:
**To the reviewer:**

The CRC24 checksum is not exposed, which is why it was easier to add this little function, rather than find a way around it - for example, accessing a private type field would be much nastier.